sklearn/kernel_approximation.py

```diff
@@ -991,6 +991,11 @@ def fit(self, X, y=None):
         self : object
             Returns the instance itself.
         """
+        xp, _ = get_namespace(X)
+        if X.ndim < 2:
+            X = xp.reshape(X, (1, -1))
+        if y is not None:
+            y = xp.asarray(y, device=device(X))
```

Comment:
Why change `y`? it is not used by this transformer (as this is an unsupervised estimator). There is no need to change it at all.